src/api/telemetry/TelemetryAPI.js

```diff
@@ -370,7 +370,8 @@ define([
     TelemetryAPI.prototype.commonValuesForHints = function (metadatas, hints) {
         var options = metadatas.map(function (metadata) {
             var values = metadata.valuesForHints(hints);
-            return _.indexBy(values, 'key');
+            const keyBy = (array, key) => (array || []).reduce((r, x) => ({ ...r, [key ? x[key] : x]: x }), {});
```

Comment:
This code is very difficult to read. Could we please rewrite it to be a little clearer?
It might be easier to read if it had more descriptive variable names, and did not use the ternary operator. I don't think we need to support the case where a key is not provided.

Reply:
Re-wrote the function in array-utils as:
```
export const flatten = (array) => array.reduce((acc, curr) => {
    return acc.concat(curr);
}, []);

```

Reply:
Did same for flatten:
```
export const flatten = (array) => array.reduce((acc, curr) => {
    return acc.concat(curr);
}, []);

```

Reply:
added flattenDeep():
```
export const flattenDeep = array => {
    return array.reduce((acc, curr) => {
        return Array.isArray(curr) ? acc.concat(flattenDeep(curr)) : acc.concat(curr);
    }, []);
};
```